### DISCUSSION TITLE
Make .md content collection pages be picked up by the sitemap integration.

### DISCUSSION THREAD

**Discussion by FlipFloop** [withastro/roadmap#841] [Proposal]
### Body

# Summary

Make `.md` content collection pages be picked up by the sitemap integration.

# Background & Motivation

These `.md` pages are not being picked up, and I don't think they should have to be defined as custom pages.

# Goals

Make it default to include content collections, with an option to disable certain pages based on a certain defined prop

# Example
```javascript
sitemap({ 
    disableMarkdownPages: False,
    disabledMdIf: draft == True,
})
 
```

**Comment by florian-lefebvre** [comment]
Your proposal is a bit confusing. Are you talking about content collections md files or md files under src/pages? Sharing how the project structure looks would help as well

**Reply by FlipFloop** [reply]
Je parle des fichiers sous "content collections" (car je pense que les fichiers .md sous `/pages` sont déjà inclus --> à vérifier).
https://docs.astro.build/en/tutorials/add-content-collections/#_top

**Reply by florian-lefebvre** [reply]
I see (just answering in English so that anyone reading this can understand properly). I don't think we can do what you suggest because content collections can be used in many ways, not only for pages. We have no way to know if `src/content/blog` is actually meant to be used as routes, and what format (/blog, /articles, just on the homepage)? It could be anything. The only way is be defining routes yourself.
Does that make sense? If not, feel free to ask other questions! (En français si tu préfères)

**Reply by FlipFloop** [reply]
Hmmmh, but when the site is built, isn't it know what pages can be referenced? How would does the static site mode work then?

**Reply by florian-lefebvre** [reply]
I feel like what you want is just achieved by using `getStaticPaths` (or using `Astro.params` in SSR). If you don't want to include some entries, just do `.filter`! Feel free to join [Astro's Discord](https://astro.build/chat) to get help for your use-case 👍
